icicle-compiler/src/Icicle/Sea/FromAvalanche/Base.hs

```diff
@@ -35,25 +40,75 @@ import           Text.Printf (printf)
 
 ------------------------------------------------------------------------
 
-textOfName :: Pretty n => n -> Text
-textOfName = T.pack . show . seaOfName
-
-seaOfName :: Pretty n => n -> Doc
-seaOfName = string . fmap mangle . show . pretty
+-- | A legal C identifier.
+newtype SeaName = SeaName {
+    getSeaName :: Text
+ } deriving (Eq, Ord, Show)
+
+-- | A value that will need be quoted in C.
+newtype SeaString = SeaString {
+    getSeaString :: Text
+ } deriving (Eq, Ord, Show)
+
+instance Pretty SeaName where
+  pretty =
+    string . Text.unpack . takeSeaName
+
+instance Pretty SeaString where
+  pretty =
+    string . Text.unpack . getSeaString
+
+takeSeaName :: SeaName -> Text
+takeSeaName = getSeaName
+
+seaNameValidHead :: Char -> Bool
+seaNameValidHead c =
+  c >= 'a' && c <= 'z'
+{-# INLINE seaNameValidHead #-}
```

Comment:
but shouldn't capital letters be allowed then? it seems like they're not allowed in the head or the tail based on these functions